### PR TITLE
fix: Bring back DefaultVerify

### PR DESCRIPTION
**Pull request by NimJay**
## Description
Please see [this GitHub pull-request for a different JSS (Jump Start Solution)](https://github.com/GoogleCloudPlatform/terraform-example-deploy-java-multizone/pull/99).

## Checklist
- [x] Please **merge** this PR for me once it is approved.

### CODE REVIEW

**Comment by glasnt** [comment]
int tests are failing because TF wants to delete the default firestore. 

This could be fixed (potentially) with a `deletion_policy` or similar attribute. 

Potentially could also be fixed with the changes in #124. Could this PR be combined with that one to see if it works together?

**Comment by NimJay** [comment]
The 2nd `terraform apply` was trying to destroy the Firestore database.
This is because `local.has_default_firestore_database` renders as true during the 2nd `terraform apply`.
So, in addition to `local.has_default_firestore_database`, we also need to check if this Terraform module was the one that created the the Firestore database.

File: infra/main.tf [hunk 1]
```diff
@@ -327,15 +327,23 @@ data "google_cloud_asset_resources_search_all" "default_firestore_database" {
...
+    || var.google_firestore_database.database[0]) # ... unless that default Firestore DB was created by this Terraform module.
+  ) ? 1 : 0
   project                     = var.project_id
   name                        = "(default)"
   location_id                 = "nam5"
   type                        = "FIRESTORE_NATIVE"
   concurrency_mode            = "PESSIMISTIC"
   app_engine_integration_mode = "DISABLED"
+  # "ABANDON" means "terraform destroy" won't actually destroy this Google Cloud resource.
+  deletion_policy = "ABANDON"
```
Comment (NimJay) [inline infra/main.tf:346]: `deletion_policy = "ABANDON"` is actually the default value. See [docs](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/firestore_database#deletion_policy).
I'm just making it more clear that the Firestore database won't be deleted during `terraform destroy`.

**Comment by NimJay** [comment]
Looks like I ran into a Terraform issue:
```
Terraform crashed! This is always indicative of a bug within Terraform.
...
panic: no expansion has been registered for module.dynamic_web_app.google_firestore_database.database
```

I need to know the Terraform verison. I just added:
```
# Log Terraform version
# Remove this step after resolving https://github.com/GoogleCloudPlatform/cloud-foundation-toolkit/issues/2321
- id: log-terraform-version
  name: 'gcr.io/cloud-foundation-cicd/$_DOCKER_IMAGE_DEVELOPER_TOOLS:$_DOCKER_TAG_VERSION_DEVELOPER_TOOLS'
  args: ['/bin/bash', '-c', 'terraform version']
```

**Pull request closed without merging**